docs/UsersGuide/dev/code-style.md

```diff
@@ -0,0 +1,123 @@
+appropriately and edit-around these modifiers as necessary. A `.clang-format` file is supplied at the root of the
+repository.
+
+### Rules of Thumb
+
+This section will document rules of thumb when developing F´ C++.  These augment the formal guidelines to help users
+develop code that fits well with the entirety of F´. 
+
+#### Documentation
+
+Components need to be documented through an Software Description Documents (SDD) placed in a `docs` sub folder. This SDD
+should capture:
+
+1. Requirements
+2. Design
+3. Implementation and Usage Notes
+
+C++ documentation is generated using doxygen. [Doxygen-style comments](https://www.doxygen.nl/manual/docblocks.html)
+should be used to annotate functions and variables. C++ APIs and libraries must be documented to produces usable doxygen
+documentation. Components should be documented in a similar fashion but the SDD captures the primary documentation for
+components.
+
+### Asserts, WARNING Events, and FATAL Events
+
+Software should operate correctly in the event of off-nominal behavior. Asserts and events can be used to handle and/or
+report these occurrences. Runtime asserts (i.e. `FW_ASSERT`) are used to ensure software integrity and assumptions.
+These are used to ensure that arguments passed to functions are within bounds, return values make since, and other
+confidence checks. Runtime assertions should never be used to validate user inputs nor results of hardware interaction
+as these may cause errors outside software control.
+
+In these cases an event should be emitted to report the error, and the software should appropriately continue execution
+even in the case of erroneous input. The severity of the event emitted should be determined from the following rules:
+
+1. `WARNING_LO` events reflect issues localized to this component where recovery is possible.
+2. `WARNING_HI` events reflect issues affecting the larger system where recovery is possible.
+3. `FATAL` events reflect issues where recovery is not possible. The system is typically restarted in this case.
+4. `WARNING_HI` and `FATAL` events must be documented in the SDD and approved as part of the submission review.
+
+#### Use of Multiple-Inheritance and Virtual Base Classes
+
+Many of F´ constructs are required to inherit from a autocoded class that then inherits from framework base classes.
+This can present a problem as this precludes inheritance from other structures. Thus limited multiple-inheritance is
+permitted. Basic guidelines are presented here, however; use of multiple-inheritance will likely be discussed during
+the submission review.
+
+1. Only use multiple-inheritance when absolutely required
+2. Avoid inheritance from more than one hand-coded class
+3. Avoid inheritance from hand-coded descendants of autocoded classes and parents (e.g. `Fw::PortBase` and
+   `Fw::PassiveComponentBase`) not. i.e avoid inheritance from other Components, Ports, or classes that do.
+
```

Comment:
4. Avoid multiple inheritance paths that derive from a common base class (the diamond problem): https://www.cprogramming.com/tutorial/virtual_inheritance.html